### PR TITLE
Migrate all but DBUser.java from 'userDB' package to 'userdb'

### PR DESCRIPTION
**Pull request by DanVanAtta**
There are compatibility problems if we move 'DBUser.java', but we are free to move the other class files.
'userDB' is a checkstyle violation for package naming, it makes working with these classes difficult. Adding a test for example will add a checkstyle violation since there is a new (test) class in a badly named package. In this update we move the classes we can to reduce this problem.

### CODE REVIEW

File: src/main/java/games/strategy/engine/lobby/server/ModeratorController.java [hunk 1]
```diff
@@ -2,12 +2,12 @@
 
 import java.util.Date;
 
-import games.strategy.engine.lobby.server.userDB.BannedMacController;
-import games.strategy.engine.lobby.server.userDB.BannedUsernameController;
+import games.strategy.engine.lobby.server.userdb.BannedMacController;
+import games.strategy.engine.lobby.server.userdb.BannedUsernameController;
 import games.strategy.engine.lobby.server.userDB.DBUser;
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/lobby/server/ModeratorController.java]: Checkstyle: "Wrong lexicographical order for 'games.strategy.engine.lobby.server.userDB.DBUser' import. Should be before 'games.strategy.engine.lobby.server.userdb.BannedUsernameController'."

File: src/main/java/games/strategy/engine/lobby/server/login/LobbyLoginValidator.java [hunk 2]
```diff
@@ -14,12 +14,12 @@
 
 import games.strategy.engine.framework.startup.ui.InGameLobbyWatcher;
 import games.strategy.engine.lobby.server.LobbyServer;
-import games.strategy.engine.lobby.server.userDB.BadWordController;
-import games.strategy.engine.lobby.server.userDB.BannedMacController;
-import games.strategy.engine.lobby.server.userDB.BannedUsernameController;
+import games.strategy.engine.lobby.server.userdb.BadWordController;
+import games.strategy.engine.lobby.server.userdb.BannedMacController;
+import games.strategy.engine.lobby.server.userdb.BannedUsernameController;
 import games.strategy.engine.lobby.server.userDB.DBUser;
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/lobby/server/login/LobbyLoginValidator.java]: Checkstyle: "Wrong lexicographical order for 'games.strategy.engine.lobby.server.userDB.DBUser' import. Should be 
 before 'games.strategy.engine.lobby.server.userdb.BannedUsernameController'."

File: src/test/java/games/strategy/engine/lobby/server/login/LobbyLoginValidatorTest.java [hunk 3]
```diff
@@ -12,10 +12,10 @@
 import org.junit.Test;
 
 import games.strategy.engine.lobby.server.LobbyServer;
-import games.strategy.engine.lobby.server.userDB.BadWordController;
+import games.strategy.engine.lobby.server.userdb.BadWordController;
 import games.strategy.engine.lobby.server.userDB.DBUser;
```
Comment (ssoloff) [inline src/test/java/games/strategy/engine/lobby/server/login/LobbyLoginValidatorTest.java]: Checkstyle: "Wrong lexicographical order for 'games.strategy.engine.lobby.server.userDB.DBUser' import. Should be before 'games.strategy.engine.lobby.server.userdb.BadWordController'."

**Comment by DanVanAtta** [comment]
> Not sure if this is going to be a problem, but the sibling packages userdb and userDB only differ by case. Is that going to be an issue for Windows devs?

In theory yes, in practice I'm pretty sure not. If it turns out to be a problem, we can fix it.

**Comment by ssoloff** [comment]
@DanVanAtta 

On Windows, the content of both `userdb` and `userDB` get merged into `userDB` when I clone:

server

userdb

And I get 16 compilation errors when running the `test` task:

```
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\login\LobbyLoginValidatorTest.java:16: error: cannot find symbol
import games.strategy.engine.lobby.server.userdb.BadWordController;
                                                ^
  symbol:   class BadWordController
  location: package games.strategy.engine.lobby.server.userdb
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\login\LobbyLoginValidatorTest.java:17: error: cannot find symbol
import games.strategy.engine.lobby.server.userdb.DbUserController;
                                                ^
  symbol:   class DbUserController
  location: package games.strategy.engine.lobby.server.userdb
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\login\LobbyLoginValidatorTest.java:18: error: cannot find symbol
import games.strategy.engine.lobby.server.userdb.HashedPassword;
                                                ^
  symbol:   class HashedPassword
  location: package games.strategy.engine.lobby.server.userdb
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\ModeratorControllerTest.java:20: error: cannot find symbol
import games.strategy.engine.lobby.server.userdb.DbUserController;
                                                ^
  symbol:   class DbUserController
  location: package games.strategy.engine.lobby.server.userdb
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\ModeratorControllerTest.java:21: error: cannot find symbol
import games.strategy.engine.lobby.server.userdb.HashedPassword;
                                                ^
  symbol:   class HashedPassword
  location: package games.strategy.engine.lobby.server.userdb
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\login\LobbyLoginValidatorTest.java:72: error: cannot find symbol
          new HashedPassword(MD5Crypt.crypt("foo")));
              ^
  symbol:   class HashedPassword
  location: class LobbyLoginValidatorTest
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\login\LobbyLoginValidatorTest.java:68: error: cannot find symbol
    new DbUserController().createUser(
        ^
  symbol:   class DbUserController
  location: class LobbyLoginValidatorTest
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\login\LobbyLoginValidatorTest.java:84: error: cannot find symbol
    new BadWordController().addBadWord("bitCh");
        ^
  symbol:   class BadWordController
  location: class LobbyLoginValidatorTest
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\login\LobbyLoginValidatorTest.java:105: error: cannot find symbol
        new HashedPassword(hashedPassword));
            ^
  symbol:   class HashedPassword
  location: class LobbyLoginValidatorTest
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\login\LobbyLoginValidatorTest.java:101: error: cannot find symbol
    new DbUserController().createUser(
        ^
  symbol:   class DbUserController
  location: class LobbyLoginValidatorTest
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\ModeratorControllerTest.java:46: error: cannot find symbol
        new HashedPassword(MD5Crypt.crypt(adminName)));
            ^
  symbol:   class HashedPassword
  location: class ModeratorControllerTest
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\ModeratorControllerTest.java:41: error: cannot find symbol
    new DbUserController().createUser(
        ^
  symbol:   class DbUserController
  location: class ModeratorControllerTest
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\userDB\BadWordControllerTest.java:14: error: cannot find symbol
    final BadWordController controller = new BadWordController();
          ^
  symbol:   class BadWordController
  location: class BadWordControllerTest
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\userDB\BadWordControllerTest.java:14: error: cannot find symbol
    final BadWordController controller = new BadWordController();
                                             ^
  symbol:   class BadWordController
  location: class BadWordControllerTest
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\userDB\HashedPasswordTest.java:23: error: cannot find symbol
            new HashedPassword(valid).isValidSyntax(), is(true)));
                ^
  symbol:   class HashedPassword
  location: class HashedPasswordTest
C:\Users\s-soloff\Downloads\triplea\src\test\java\games\strategy\engine\lobby\server\userDB\HashedPasswordTest.java:37: error: cannot find symbol
            new HashedPassword(invalid).isValidSyntax(), is(false)));
                ^
  symbol:   class HashedPassword
  location: class HashedPasswordTest
```

**Comment by ron-murhammer** [comment]
Yeah, I don't think you can have packages with the same name just different case on windows.

**Comment by DanVanAtta** [comment]
Renamed the package to address the last concern, merging now so we can move on to dependent branches.

**Pull request merged**